src/components/Common/JsonEditor.vue

```diff
@@ -26,70 +27,77 @@
 </style>
 
 <script>
-import Vue from 'vue'
-let editor
+import Vue from 'vue';
 
 export default {
   name: 'JsonEditor',
   props: {
     content: String,
+    id: {
+      type: String,
+      default: Date.now().toString() + Math.random().toString()
+    },
     myclass: {
       type: String,
       default: ''
     },
     readonly: Boolean,
-    id: String,
     height: { type: Number, default: 250 }
   },
+  data() {
+    return {
+      editor: null
```

Comment:
What is the purpose of putting the `editor` variable in the component state?

Reply:
it is necessary to have the `editor` variable scoped to the component to be able to use several instances

Reply:
why?

Reply:
If we let the `editor` in global, if we add multiple json-editors there will still be only one `editor` variable wich will be overrided each time we instanciate a new json-editor.
So call setContent method of a json-editor component will always set the content on the last instanciated editor no matter wich json-editor's method we call

Reply:
I see. The `editor` variable is scoped to the JSONEditor.vue module, which is a singleton, in the CommonJS modules model. The module exports a VueJS Component config, which is passed to the `Vue` constructor each time this component is instantiated. But this config has a reference to the `editor` variable, which is actually overwritten, as you rightly say, each time the component is instantiated. By putting the variable inside the `data` structure, you make it local to the component instead, which is obviously not a singleton. This is a nice thing to keep in mind for future developments.

I still don't like the idea of putting a complex object (i.e. one that has attributes and functions) inside the `data` structure, because it is meant to keep track of the _state_ of the component, yet this object is just logic to make the editor work. We should find a way to replicate this reference each time the component is created, and avoid the reference to be overwritten.

I am not really hurried to ship this fix, but if you are, we can keep this enhancement for later. What do you think?

Reply:
After merge this PR I'll create an issue concerning the refacto of the json-editor and we'll probably need a workshop